src/git.rs

```diff
@@ -1,50 +1,40 @@
 pub mod git {
-    use std::io::Error;
+    use std::io::Result;
     use std::process::Command;
+    use utils::utils;
 
-    pub fn git_commit_and_push(repo_path: &String, msg: String) -> Result<(), ()> {
-        // TODO: See how to chain these function calls
-        git_add(repo_path).unwrap();
-        git_commit(repo_path, msg).unwrap();
-        git_push(repo_path).unwrap();
-        Ok(())
+    pub fn git_commit_and_push(repo_path: &String, msg: String) -> Result<()> {
+        git_add(repo_path)
+            .and(git_commit(repo_path, msg))
+            .and(git_push(repo_path))
     }
 
-    fn get_git_path() -> Result<String, ()> {
-        // TODO: Do not have it hard-coded, look for it in common places
-        Ok(String::from("/usr/bin/git"))
-    }
-
-    fn git_add(repo_path: &String) -> Result<(), Error> {
-        let git = get_git_path().unwrap(); // TODO
-        match Command::new(git)
-            .arg(format!("--git-dir={}/.git/", repo_path))
-            .arg(format!("--work-tree={}", repo_path))
+    fn git_add(repo_path: &String) -> Result<()> {
+        match Command::new(git())
+            .args(default_args(repo_path).iter())
             .arg("add")
-            .arg("-A")
+            .arg("./README.md")
```

Comment:
this is not really portable

Reply:
Could you expand on this?

Reply:
nevermind, I thought it may not work on windows, but it seems it works according to the documentation.